code/datums/emergency_calls/emergency_call.dm

```diff
@@ -0,0 +1,217 @@
+
+	var/list/total_calls = typesof(/datum/emergency_call)
+	if(!length(total_calls))
+		CRASH("No distress Datums found.")
+
+	for(var/x in total_calls)
+		var/datum/emergency_call/D = new x()
+		if(!D?.name)
+			continue //The default parent, don't add it
+		all_calls += D
+
+
+//Randomizes and chooses a call datum.
+/datum/game_mode/proc/get_random_call()
+	var/datum/emergency_call/chosen_call
+	var/list/valid_calls = list()
+
+	for(var/datum/emergency_call/E in all_calls) //Loop through all potential candidates
+		if(E.probability < 1) //Those that are meant to be admin-only
+			continue
+
+		valid_calls.Add(E)
+
+		if(prob(E.probability))
+			chosen_call = E
+			break
+
+	if(!istype(chosen_call))
+		chosen_call = pick(valid_calls)
+
+	return chosen_call
+
+/datum/emergency_call/proc/show_join_message()
+	if(!mob_max || !ticker?.mode) //Not a joinable distress call.
+		return
+
+	for(var/i in GLOB.player_list)
+		if(!isghost(i)) continue
+		var/client/M = i
+		to_chat(M, "<br><font size='3'><span class='attack'>An emergency beacon has been activated. Use the <B>IC > <a href='byond://?src=[REF(M)];join_ert=1'>Join Response Team</a></b> verb to join!</span></font><br>")
+		to_chat(M, "<span class='attack'>You cannot join if you have Ghosted before this message.</span><br>")
+
+
+/datum/game_mode/proc/activate_distress(datum/emergency_call/chosen_call)
+	picked_call = chosen_call || get_random_call()
+
+	if(ticker?.mode?.waiting_for_candidates) //It's already been activated
+		return FALSE
+
+	picked_call.mob_max = rand(5, 15)
+
+	picked_call.activate()
+
+/datum/emergency_call/proc/reset()
+	if(candidate_timer)
+		deltimer(candidate_timer)
+		candidate_timer = null
+	if(cooldown_timer)
+		deltimer(cooldown_timer)
+		cooldown_timer = null
+	members.Cut()
+	candidates.Cut()
+	ticker.mode.waiting_for_candidates = FALSE
+	ticker.mode.on_distress_cooldown = FALSE
+	message_admins("Distress beacon: [name] has been reset.")
+
+/datum/emergency_call/proc/activate(announce = TRUE)
+	if(!ticker?.mode) //Something horribly wrong with the gamemode ticker
+		message_admins("Distress beacon: [name] attempted to activate but no gamemode exists")
+		return FALSE
+
+	if(ticker.mode.on_distress_cooldown) //It's already been called.
+		message_admins("Distress beacon: [name] attempted to activate but distress is on cooldown")
+		return FALSE
+
+	if(mob_max > 0)
+		ticker.mode.waiting_for_candidates = TRUE
+
+	show_join_message() //Show our potential candidates the message to let them join.
+	message_admins("Distress beacon: '[name]' activated. Looking for candidates.")
+
+	if(announce)
+		pr_announce.Announce("A distress beacon has been launched from the USG Ishimura.", "Priority Alert")
+
+	ticker.mode.on_distress_cooldown = TRUE
+
+	candidate_timer = addtimer(CALLBACK(src, .proc/do_activate, announce), 5 MINUTES, TIMER_STOPPABLE)
+
+/datum/emergency_call/proc/do_activate(announce = TRUE)
+	candidate_timer = null
+	ticker.mode.waiting_for_candidates = FALSE
```

Comment:
Did you add this variable? If so, it seems a bit ambiguously named

if not, then its probably intended for antag selection, this might not be a good way to use it

Reply:
Add.

Reply:
why is this tied to gamemode at all? What if we want to do multiple emergency calls?